builder/builder-cache.c

```diff
@@ -80,6 +80,10 @@ builder_cache_finalize (GObject *object)
   g_free (self->last_parent);
   if (self->unused_stages)
     g_hash_table_unref (self->unused_stages);
+  if (self->stage)
+    g_free (self->stage);
+  if (self->repo)
+    g_object_unref (self->repo);
```

Comment:
`g_free()` is safe against NULL values.

You could've used `g_clear_object()` and avoided the second if as well.